src/core/ledger/ledger.js

```diff
@@ -575,6 +602,58 @@ export class Ledger {
     account.identity = updatedIdentity;
   }
 
+  /**
+   * setPayoutAddress allows participants to set a payable address to collect
+   * grain. These addresses are keyed on a specific currency, which ensures that
+   * users don't erroneously receive a grain distribution to an address that
+   * cannot handle it (such as a custodial wallet, or rigidly-designed
+   * contract) and effectively lose that reward.
+   *
+   * An address may be deleted by passing in an empty string for the
+   * `address` parameter. This is useful in case the underlying private key
+   * is compromised or the exchange hosting a custodial account is hacked.
+   */
+  setPayoutAddress(
+    id: IdentityId,
+    payoutAddress: PayoutAddress | null,
+    chainId: ChainId,
+    currencyAddress: EthAddress | void
+  ): Ledger {
+    this._createAndProcessEvent({
+      type: "SET_PAYOUT_ADDRESS",
+      id,
+      payoutAddress,
+      currencyId: {chainId, currencyAddress},
+    });
```

Comment:
Would be cleaner to take in the `Currency` object type I described above instead, something like this:
```ts
  setPayoutAddress(
     id: IdentityId,
     currency: Currency,
     payoutAddress: PayoutAddress | null,
  ): Ledger {
    this._createAndProcessEvent({
      type: "SET_PAYOUT_ADDRESS",
      id,
      currency,
      payoutAddress,
    });
```

Also, this structure implies that different people would be able to get payouts in different tokens on different chains. Is there any use case where you think that might be relevant? Feels redundant to have the currency object in every ledger entry for setting a payout address.

Reply:
@hammadj I considered this but I feel like this is leaking implementation details. Consumers of this API to only need to pass in `ChainId`s and and and an optional ethereum address to generate a `CurrencyId` internally.

Reply:
Wouldn't the `symbol` be a valuable thing to have in there as well, especially for rendering user-readable ledger history and other UI elements?